In [1]:
import pandas as pd

In [2]:
Dataset=pd.read_csv('Medataset1.csv')

In [3]:
Dataset.head()

Unnamed: 0     ID    BIRTHDAY NAME      DIPLOMTYPE  DIPLOMPERCENTAGE  \
0           0   9867  1995-05-06    H  DIPLÔME D'ETAT              52.0   
1           1   9867  1995-05-06    H  DIPLÔME D'ETAT              52.0   
2           2   9891  1996-01-01    F  DIPLÔME D'ETAT              50.0   
3           3   9891  1996-01-01    F  DIPLÔME D'ETAT              50.0   
4           4  10010  1994-08-28    F  DIPLÔME D'ETAT              60.0   

  DIPLOMSECTION        DIPLOMOPTION               SCHOOL SCHOOLPROVINCE  \
0  Scientifique     Bioligie-Chimie                 Goma      NORD-KIVU   
1  Scientifique     Bioligie-Chimie                 Goma      NORD-KIVU   
2     Technique             Sociale  Lycée Sainte Ursule      NORD-KIVU   
3     Technique             Sociale  Lycée Sainte Ursule      NORD-KIVU   
4   Pédagogique  Pédagogie générale             Metanoia      NORD-KIVU   

  SCHOOLSTATUS   ACADYEAR      PERC1                  MENT1      PERC2  \
0      inconnu  2014-2015        NaN  ASSIMILE AUX AJOURNES  59.200001   
1      inconnu  2015-2016        NaN  ASSIMILE AUX AJOURNES        NaN   
2      inconnu  2014-2015  50.599998                AJOURNE  56.299999   
3      inconnu  2015-2016  51.500000                AJOURNE  55.500000   
4      inconnu  2014-2015        NaN  ASSIMILE AUX AJOURNES  59.500000   

          MENT2                  FAC      OPT PROM  
0  SATISFACTION  Faculté de Médecine  Bio-Méd   G1  
1       AJOURNE  Faculté de Médecine  Bio-Méd   G2  
2  SATISFACTION  Faculté de Médecine  Bio-Méd   G1  
3  SATISFACTION  Faculté de Médecine  Bio-Méd   G2  
4  SATISFACTION  Faculté de Médecine  Bio-Méd   G1

In [4]:
DatasetSMF=pd.read_csv('MedecineDataSemiFInished.csv')

In [5]:
DatasetSMF

Unnamed: 0     ID    BIRTHDAY NAME      DIPLOMTYPE  DIPLOMPERCENTAGE  \
0             0   1670  1973-05-12    H         inconnu              51.0   
1             1   9841  1996-11-23    H  DIPLÔME D'ETAT              53.0   
2             2   9867  1995-05-06    H  DIPLÔME D'ETAT              52.0   
3             3   9879  1997-11-22    F  DIPLÔME D'ETAT              62.0   
4             4   9887  1995-07-17    H  DIPLÔME D'ETAT              59.0   
5             5   9891  1996-01-01    F  DIPLÔME D'ETAT              50.0   
6             6   9909  1994-11-04    H  DIPLÔME D'ETAT              51.0   
7             7   9915  1996-06-25    F  DIPLÔME D'ETAT              64.0   
8             8   9942  1996-01-21    F  DIPLÔME D'ETAT              68.0   
9             9   9950  1994-03-03    H  DIPLÔME D'ETAT              60.0   
10           10   9965  1992-02-05    H  DIPLÔME D'ETAT              58.0   
11           11   9975  1994-11-24    H  DIPLÔME D'ETAT              60.0   
12           12   9978  1996-06-11    H  DIPLÔME D'ETAT              65.0   
13           13   9983  1997-03-26    H  DIPLÔME D'ETAT              65.0   
14           14   9993  1993-07-26    H  DIPLÔME D'ETAT              61.0   
15           15  10001  1993-10-10    H  DIPLÔME D'ETAT              58.0   
16           16  10008  1988-05-15    H  DIPLÔME D'ETAT              63.0   
17           17  10010  1994-08-28    F  DIPLÔME D'ETAT              60.0   
18           18  10022  1997-09-13    H  DIPLÔME D'ETAT              59.0   
19           19  10037  1996-04-10    H  DIPLÔME D'ETAT              58.0   
20           20  10048  1996-10-22    H  DIPLÔME D'ETAT              60.0   
21           21  10090  1994-10-04    F  DIPLÔME D'ETAT              53.0   
22           22  10093  1996-04-06    F  DIPLÔME D'ETAT              53.0   
23           23  10096  1997-01-30    H  DIPLÔME D'ETAT              66.0   
24           24  10114  1995-09-29    F  DIPLÔME D'ETAT              51.0   
25           25  10116  1997-01-05    F  DIPLÔME D'ETAT              77.0   
26           26  10128  1994-12-25    F  DIPLÔME D'ETAT              58.0   
27           27  10150  1991-03-22    H  DIPLÔME D'ETAT              61.0   
28           28  10155  1995-01-03    H  DIPLÔME D'ETAT              51.0   
29           29  10159  1995-07-21    H  DIPLÔME D'ETAT               0.0   
..          ...    ...         ...  ...             ...               ...   
164         164  11828  1997-12-25    F  DIPLÔME D'ETAT              57.0   
165         165  11834  1997-03-09    F  DIPLÔME D'ETAT              64.0   
166         166  11842  1997-02-06    H  DIPLÔME D'ETAT              55.0   
167         167  11850  1992-12-06    H  DIPLÔME D'ETAT              52.0   
168         168  11868  1996-08-08    F  DIPLÔME D'ETAT              67.0   
169         169  11897  1998-01-12    F  DIPLÔME D'ETAT              51.0   
170         170  11978  1995-08-08    F  DIPLÔME D'ETAT              61.0   
171         171  11997  1997-01-24    H  DIPLÔME D'ETAT              61.0   
172         172  12013  1997-10-25    F  DIPLÔME D'ETAT              51.0   
173         173  12039  1996-09-20    F  DIPLÔME D'ETAT              51.0   
174         174  12051  1997-02-02    H  DIPLÔME D'ETAT              72.0   
175         175  12062  1995-04-30    F  DIPLÔME D'ETAT              52.0   
176         176  12069  1997-02-06    H  DIPLÔME D'ETAT              60.0   
177         177  12070  1998-10-06    F  DIPLÔME D'ETAT              61.0   
178         178  12082  1995-09-07    F  DIPLÔME D'ETAT              51.0   
179         179  12119  1996-02-02    F  DIPLÔME D'ETAT              59.0   
180         180  12120  1976-08-26    F  DIPLÔME D'ETAT              52.0   
181         181  12132  1995-03-28    H  DIPLÔME D'ETAT              58.0   
182         182  12140  1995-03-13    H  DIPLÔME D'ETAT              57.0   
183         183  12143  1993-12-05    H  DIPLÔME D'ETAT              53.0   
1

# Data Exploration and Preparation

In this notebook we will follow all the process of data preparation and exploration and we will follow this tutorial from datacamp
https://www.datacamp.com/community/tutorials/exploratory-data-analysis-python

first let start by the process called : Exploratory Data Analysis (EDA) m

## Exploratory Data Analysis (EDA)

we will try to sumarise data with the descriptive statisticsm

### EDA And Data Mining (DM)

those fiels have the used the some methods but are diffrents DM is application oriented while EDA just find to underline the relationship between differents feature 

the first task for EDA is to describe our dataset so let do it:

In [6]:
DatasetSMF.describe()

Unnamed: 0            ID  DIPLOMPERCENTAGE  PERC1-2ndYear  PERC1-1year  \
count  194.000000    194.000000        194.000000      90.000000    90.000000   
mean    96.500000  10830.190722         59.912371      54.227778    54.004444   
std     56.147128    978.879337         43.280715       3.552054     3.708809   
min      0.000000   1670.000000          0.000000      42.700001    42.700001   
25%     48.250000  10285.250000         52.000000      52.000000    51.325000   
50%     96.500000  10626.500000         57.000000      54.750000    54.450001   
75%    144.750000  11598.250000         61.000000      56.574999    56.574999   
max    193.000000  12252.000000        645.000000      63.799999    63.799999   

       PERC2-2ndYear  PERC2-1year  ACADYEAR-NUMBER  
count     163.000000   163.000000       194.000000  
mean       58.162577    59.138650         1.340206  
std         4.435520     4.226362         0.475004  
min        42.000000    42.000000         1.000000  
25%        55.950001    56.950001         1.000000  
50%        58.599998    59.500000         1.000000  
75%        61.350000    61.700001         2.000000  
max        70.300003    70.300003         2.000000

In [7]:
#DatasetSMF.rename(columns={0:'Unnamed: 0'},inplace=False)

In [8]:
DatasetSMF.describe()

ID            ID  DIPLOMPERCENTAGE  PERC1-2ndYear  PERC1-1year  \
count  194.000000    194.000000        194.000000      90.000000    90.000000   
mean    96.500000  10830.190722         59.912371      54.227778    54.004444   
std     56.147128    978.879337         43.280715       3.552054     3.708809   
min      0.000000   1670.000000          0.000000      42.700001    42.700001   
25%     48.250000  10285.250000         52.000000      52.000000    51.325000   
50%     96.500000  10626.500000         57.000000      54.750000    54.450001   
75%    144.750000  11598.250000         61.000000      56.574999    56.574999   
max    193.000000  12252.000000        645.000000      63.799999    63.799999   

       PERC2-2ndYear  PERC2-1year  ACADYEAR-NUMBER  
count     163.000000   163.000000       194.000000  
mean       58.162577    59.138650         1.340206  
std         4.435520     4.226362         0.475004  
min        42.000000    42.000000         1.000000  
25%        55.950001    56.950001         1.000000  
50%        58.599998    59.500000         1.000000  
75%        61.350000    61.700001         2.000000  
max        70.300003    70.300003         2.000000

with this comand we discover some useful informations about the data such as the mean of diplome percentage is 59.9 %
we also discover some outlier and we will treat it later (min dip percentage is 0 and max is 645 that is impossible) we also discover that the max percentrage in 1rst session is 63% andin 2nd session it's 70% 

let delete a duplicate colum id

In [17]:
#DatasetSMF.drop(axis=1,labels=DatasetSMF.columns[0])

BIRTHDAY NAME      DIPLOMTYPE  DIPLOMPERCENTAGE DIPLOMSECTION  \
0    1973-05-12    H         inconnu              51.0     TECHNIQUE   
1    1996-11-23    H  DIPLÔME D'ETAT              53.0  Scientifique   
2    1995-05-06    H  DIPLÔME D'ETAT              52.0  Scientifique   
3    1997-11-22    F  DIPLÔME D'ETAT              62.0     Technique   
4    1995-07-17    H  DIPLÔME D'ETAT              59.0  Scientifique   
5    1996-01-01    F  DIPLÔME D'ETAT              50.0     Technique   
6    1994-11-04    H  DIPLÔME D'ETAT              51.0  Scientifique   
7    1996-06-25    F  DIPLÔME D'ETAT              64.0  Scientifique   
8    1996-01-21    F  DIPLÔME D'ETAT              68.0    Littéraire   
9    1994-03-03    H  DIPLÔME D'ETAT              60.0  Scientifique   
10   1992-02-05    H  DIPLÔME D'ETAT              58.0  Scientifique   
11   1994-11-24    H  DIPLÔME D'ETAT              60.0  Scientifique   
12   1996-06-11    H  DIPLÔME D'ETAT              65.0  Scientifique   
13   1997-03-26    H  DIPLÔME D'ETAT              65.0  Scientifique   
14   1993-07-26    H  DIPLÔME D'ETAT              61.0     Technique   
15   1993-10-10    H  DIPLÔME D'ETAT              58.0  Scientifique   
16   1988-05-15    H  DIPLÔME D'ETAT              63.0   Pédagogique   
17   1994-08-28    F  DIPLÔME D'ETAT              60.0   Pédagogique   
18   1997-09-13    H  DIPLÔME D'ETAT              59.0  Scientifique   
19   1996-04-10    H  DIPLÔME D'ETAT              58.0  Scientifique   
20   1996-10-22    H  DIPLÔME D'ETAT              60.0  Scientifique   
21   1994-10-04    F  DIPLÔME D'ETAT              53.0     Technique   
22   1996-04-06    F  DIPLÔME D'ETAT              53.0    Littéraire   
23   1997-01-30    H  DIPLÔME D'ETAT              66.0  Scientifique   
24   1995-09-29    F  DIPLÔME D'ETAT              51.0  Scientifique   
25   1997-01-05    F  DIPLÔME D'ETAT              77.0  Scientifique   
26   1994-12-25    F  DIPLÔME D'ETAT              58.0   Pédagogique   
27   1991-03-22    H  DIPLÔME D'ETAT              61.0   Pédagogique   
28   1995-01-03    H  DIPLÔME D'ETAT              51.0  Scientifique   
29   1995-07-21    H  DIPLÔME D'ETAT               0.0     Technique   
..          ...  ...             ...               ...           ...   
164  1997-12-25    F  DIPLÔME D'ETAT              57.0     PEDAGOGIE   
165  1997-03-09    F  DIPLÔME D'ETAT              64.0     PEDAGOGIE   
166  1997-02-06    H  DIPLÔME D'ETAT              55.0    TECHNIQUES   
167  1992-12-06    H  DIPLÔME D'ETAT              52.0  SCIENTIFIQUE   
168  1996-08-08    F  DIPLÔME D'ETAT              67.0     PEGAGOGIE   
169  1998-01-12    F  DIPLÔME D'ETAT              51.0   COMMERCIALE   
170  1995-08-08    F  DIPLÔME D'ETAT              61.0     PEGAGOGIE   
171  1997-01-24    H  DIPLÔME D'ETAT              61.0     PEGAGOGIE   
172  1997-10-25    F  DIPLÔME D'ETAT              51.0   COMMERCIALE   
173  1996-09-20    F  DIPLÔME D'ETAT              51.0     PEDAGOGIE   
174  1997-02-02    H  DIPLÔME D'ETAT              72.0  SCIENTIFIQUE   
175  1995-04-30    F  DIPLÔME D'ETAT              52.0  SCIENTIFIQUE   
176  1997-02-06    H  DIPLÔME D'ETAT              60.0    LITTERAIRE   
177  1998-10-06    F  DIPLÔME D'ETAT              61.0     TECHNIQUE   
178  1995-09-07    F  DIPLÔME D'ETAT              51.0     TECHNIQUE   
179  1996-02-02    F  DIPLÔME D'ETAT              59.0  SCIENTIFIQUE   
180  1976-08-26    F  DIPLÔME D'ETAT              52.0     PEDAGOGIE   
181  1995-03-28    H  DIPLÔME D'ETAT              58.0     PEDAGOGIE   
182  1995-03-13    H  DIPLÔME D'ETAT              57.0     PEDAGOGIE   
183  1993-12-05    H  DIPLÔME D'ETAT              53.0     PEDAGOGIE   
184  1997-03-29    F  DIPLÔME D'ETAT              64.0     TECHNIQUE   
185  1995-09-29    F  DIPLÔME D'ETAT              58.0     TECHNIQUE   
186  1997-01-01    H  DIPLÔME D'ETAT              56.0  SCIENTIFIQUE   
187  1992-10-29    H  DIPLÔME D'ETAT           

In [29]:
#columNameOld=list(DatasetSMF.columns)

In [24]:
#columName[0]='IDFake'

In [30]:
#columName

['IDFake',
 'ID',
 'BIRTHDAY',
 'NAME',
 'DIPLOMTYPE',
 'DIPLOMPERCENTAGE',
 'DIPLOMSECTION',
 'DIPLOMOPTION',
 'SCHOOL',
 'SCHOOLPROVINCE',
 'SCHOOLSTATUS',
 'MENT1-2ndYear',
 'MENT1-1year',
 'OPT-first',
 'PROM-START',
 'PROM-END',
 'MENT2-2ndYear',
 'MENT2-1year',
 'PERC1-2ndYear',
 'PERC1-1year',
 'FAC-first',
 'PERC2-2ndYear',
 'PERC2-1year',
 'ACADYEAR-START',
 'ACADYEAR-END',
 'ACADYEAR-NUMBER']

In [35]:
#dict(zip(columNameOld,columName))

{'ACADYEAR-END': 'ACADYEAR-END',
 'ACADYEAR-NUMBER': 'ACADYEAR-NUMBER',
 'ACADYEAR-START': 'ACADYEAR-START',
 'BIRTHDAY': 'BIRTHDAY',
 'DIPLOMOPTION': 'DIPLOMOPTION',
 'DIPLOMPERCENTAGE': 'DIPLOMPERCENTAGE',
 'DIPLOMSECTION': 'DIPLOMSECTION',
 'DIPLOMTYPE': 'DIPLOMTYPE',
 'FAC-first': 'FAC-first',
 'ID': 'ID',
 'MENT1-1year': 'MENT1-1year',
 'MENT1-2ndYear': 'MENT1-2ndYear',
 'MENT2-1year': 'MENT2-1year',
 'MENT2-2ndYear': 'MENT2-2ndYear',
 'NAME': 'NAME',
 'OPT-first': 'OPT-first',
 'PERC1-1year': 'PERC1-1year',
 'PERC1-2ndYear': 'PERC1-2ndYear',
 'PERC2-1year': 'PERC2-1year',
 'PERC2-2ndYear': 'PERC2-2ndYear',
 'PROM-END': 'PROM-END',
 'PROM-START': 'PROM-START',
 'SCHOOL': 'SCHOOL',
 'SCHOOLPROVINCE': 'SCHOOLPROVINCE',
 'SCHOOLSTATUS': 'SCHOOLSTATUS'}

In [50]:
#DatasetSMF.rename(columns=dict(zip(DatasetSMF.columns[1:],columName)),inplace=False)

IDGud  IDGud    BIRTHDAY NAME      DIPLOMTYPE  DIPLOMPERCENTAGE  \
0        0   1670  1973-05-12    H         inconnu              51.0   
1        1   9841  1996-11-23    H  DIPLÔME D'ETAT              53.0   
2        2   9867  1995-05-06    H  DIPLÔME D'ETAT              52.0   
3        3   9879  1997-11-22    F  DIPLÔME D'ETAT              62.0   
4        4   9887  1995-07-17    H  DIPLÔME D'ETAT              59.0   
5        5   9891  1996-01-01    F  DIPLÔME D'ETAT              50.0   
6        6   9909  1994-11-04    H  DIPLÔME D'ETAT              51.0   
7        7   9915  1996-06-25    F  DIPLÔME D'ETAT              64.0   
8        8   9942  1996-01-21    F  DIPLÔME D'ETAT              68.0   
9        9   9950  1994-03-03    H  DIPLÔME D'ETAT              60.0   
10      10   9965  1992-02-05    H  DIPLÔME D'ETAT              58.0   
11      11   9975  1994-11-24    H  DIPLÔME D'ETAT              60.0   
12      12   9978  1996-06-11    H  DIPLÔME D'ETAT              65.0   
13      13   9983  1997-03-26    H  DIPLÔME D'ETAT              65.0   
14      14   9993  1993-07-26    H  DIPLÔME D'ETAT              61.0   
15      15  10001  1993-10-10    H  DIPLÔME D'ETAT              58.0   
16      16  10008  1988-05-15    H  DIPLÔME D'ETAT              63.0   
17      17  10010  1994-08-28    F  DIPLÔME D'ETAT              60.0   
18      18  10022  1997-09-13    H  DIPLÔME D'ETAT              59.0   
19      19  10037  1996-04-10    H  DIPLÔME D'ETAT              58.0   
20      20  10048  1996-10-22    H  DIPLÔME D'ETAT              60.0   
21      21  10090  1994-10-04    F  DIPLÔME D'ETAT              53.0   
22      22  10093  1996-04-06    F  DIPLÔME D'ETAT              53.0   
23      23  10096  1997-01-30    H  DIPLÔME D'ETAT              66.0   
24      24  10114  1995-09-29    F  DIPLÔME D'ETAT              51.0   
25      25  10116  1997-01-05    F  DIPLÔME D'ETAT              77.0   
26      26  10128  1994-12-25    F  DIPLÔME D'ETAT              58.0   
27      27  10150  1991-03-22    H  DIPLÔME D'ETAT              61.0   
28      28  10155  1995-01-03    H  DIPLÔME D'ETAT              51.0   
29      29  10159  1995-07-21    H  DIPLÔME D'ETAT               0.0   
..     ...    ...         ...  ...             ...               ...   
164    164  11828  1997-12-25    F  DIPLÔME D'ETAT              57.0   
165    165  11834  1997-03-09    F  DIPLÔME D'ETAT              64.0   
166    166  11842  1997-02-06    H  DIPLÔME D'ETAT              55.0   
167    167  11850  1992-12-06    H  DIPLÔME D'ETAT              52.0   
168    168  11868  1996-08-08    F  DIPLÔME D'ETAT              67.0   
169    169  11897  1998-01-12    F  DIPLÔME D'ETAT              51.0   
170    170  11978  1995-08-08    F  DIPLÔME D'ETAT              61.0   
171    171  11997  1997-01-24    H  DIPLÔME D'ETAT              61.0   
172    172  12013  1997-10-25    F  DIPLÔME D'ETAT              51.0   
173    173  12039  1996-09-20    F  DIPLÔME D'ETAT              51.0   
174    174  12051  1997-02-02    H  DIPLÔME D'ETAT              72.0   
175    175  12062  1995-04-30    F  DIPLÔME D'ETAT              52.0   
176    176  12069  1997-02-06    H  DIPLÔME D'ETAT              60.0   
177    177  12070  1998-10-06    F  DIPLÔME D'ETAT              61.0   
178    178  12082  1995-09-07    F  DIPLÔME D'ETAT              51.0   
179    179  12119  1996-02-02    F  DIPLÔME D'ETAT              59.0   
180    180  12120  1976-08-26    F  DIPLÔME D'ETAT              52.0   
181    181  12132  1995-03-28    H  DIPLÔME D'ETAT              58.0   
182    182  12140  1995-03-13    H  DIPLÔME D'ETAT              57.0   
183    183  12143  1993-12-05    H  DIPLÔME D'ETAT              53.0   
184    184  12145  1997-03-29    F  DIPLÔME D'ETAT              64.0   
185    185  12146  1995-09-29    F  DIPLÔME D'ETAT              58.0   
186    186  12148  1997-01-01    H  DIPLÔME D'ETAT              56.0   
187    187  12164  1992-10-29    H  DIPLÔME D

In [39]:
#del columName[0]

In [42]:
#columName[0]='IDGud'

In [46]:
#len(columNameOld)

26

In [48]:
#len(columName)

25

In [49]:
#columName

['IDGud',
 'BIRTHDAY',
 'NAME',
 'DIPLOMTYPE',
 'DIPLOMPERCENTAGE',
 'DIPLOMSECTION',
 'DIPLOMOPTION',
 'SCHOOL',
 'SCHOOLPROVINCE',
 'SCHOOLSTATUS',
 'MENT1-2ndYear',
 'MENT1-1year',
 'OPT-first',
 'PROM-START',
 'PROM-END',
 'MENT2-2ndYear',
 'MENT2-1year',
 'PERC1-2ndYear',
 'PERC1-1year',
 'FAC-first',
 'PERC2-2ndYear',
 'PERC2-1year',
 'ACADYEAR-START',
 'ACADYEAR-END',
 'ACADYEAR-NUMBER']

In [51]:
DatasetSMF

ID     ID    BIRTHDAY NAME      DIPLOMTYPE  DIPLOMPERCENTAGE  \
0      0   1670  1973-05-12    H         inconnu              51.0   
1      1   9841  1996-11-23    H  DIPLÔME D'ETAT              53.0   
2      2   9867  1995-05-06    H  DIPLÔME D'ETAT              52.0   
3      3   9879  1997-11-22    F  DIPLÔME D'ETAT              62.0   
4      4   9887  1995-07-17    H  DIPLÔME D'ETAT              59.0   
5      5   9891  1996-01-01    F  DIPLÔME D'ETAT              50.0   
6      6   9909  1994-11-04    H  DIPLÔME D'ETAT              51.0   
7      7   9915  1996-06-25    F  DIPLÔME D'ETAT              64.0   
8      8   9942  1996-01-21    F  DIPLÔME D'ETAT              68.0   
9      9   9950  1994-03-03    H  DIPLÔME D'ETAT              60.0   
10    10   9965  1992-02-05    H  DIPLÔME D'ETAT              58.0   
11    11   9975  1994-11-24    H  DIPLÔME D'ETAT              60.0   
12    12   9978  1996-06-11    H  DIPLÔME D'ETAT              65.0   
13    13   9983  1997-03-26    H  DIPLÔME D'ETAT              65.0   
14    14   9993  1993-07-26    H  DIPLÔME D'ETAT              61.0   
15    15  10001  1993-10-10    H  DIPLÔME D'ETAT              58.0   
16    16  10008  1988-05-15    H  DIPLÔME D'ETAT              63.0   
17    17  10010  1994-08-28    F  DIPLÔME D'ETAT              60.0   
18    18  10022  1997-09-13    H  DIPLÔME D'ETAT              59.0   
19    19  10037  1996-04-10    H  DIPLÔME D'ETAT              58.0   
20    20  10048  1996-10-22    H  DIPLÔME D'ETAT              60.0   
21    21  10090  1994-10-04    F  DIPLÔME D'ETAT              53.0   
22    22  10093  1996-04-06    F  DIPLÔME D'ETAT              53.0   
23    23  10096  1997-01-30    H  DIPLÔME D'ETAT              66.0   
24    24  10114  1995-09-29    F  DIPLÔME D'ETAT              51.0   
25    25  10116  1997-01-05    F  DIPLÔME D'ETAT              77.0   
26    26  10128  1994-12-25    F  DIPLÔME D'ETAT              58.0   
27    27  10150  1991-03-22    H  DIPLÔME D'ETAT              61.0   
28    28  10155  1995-01-03    H  DIPLÔME D'ETAT              51.0   
29    29  10159  1995-07-21    H  DIPLÔME D'ETAT               0.0   
..   ...    ...         ...  ...             ...               ...   
164  164  11828  1997-12-25    F  DIPLÔME D'ETAT              57.0   
165  165  11834  1997-03-09    F  DIPLÔME D'ETAT              64.0   
166  166  11842  1997-02-06    H  DIPLÔME D'ETAT              55.0   
167  167  11850  1992-12-06    H  DIPLÔME D'ETAT              52.0   
168  168  11868  1996-08-08    F  DIPLÔME D'ETAT              67.0   
169  169  11897  1998-01-12    F  DIPLÔME D'ETAT              51.0   
170  170  11978  1995-08-08    F  DIPLÔME D'ETAT              61.0   
171  171  11997  1997-01-24    H  DIPLÔME D'ETAT              61.0   
172  172  12013  1997-10-25    F  DIPLÔME D'ETAT              51.0   
173  173  12039  1996-09-20    F  DIPLÔME D'ETAT              51.0   
174  174  12051  1997-02-02    H  DIPLÔME D'ETAT              72.0   
175  175  12062  1995-04-30    F  DIPLÔME D'ETAT              52.0   
176  176  12069  1997-02-06    H  DIPLÔME D'ETAT              60.0   
177  177  12070  1998-10-06    F  DIPLÔME D'ETAT              61.0   
178  178  12082  1995-09-07    F  DIPLÔME D'ETAT              51.0   
179  179  12119  1996-02-02    F  DIPLÔME D'ETAT              59.0   
180  180  12120  1976-08-26    F  DIPLÔME D'ETAT              52.0   
181  181  12132  1995-03-28    H  DIPLÔME D'ETAT              58.0   
182  182  12140  1995-03-13    H  DIPLÔME D'ETAT              57.0   
183  183  12143  1993-12-05    H  DIPLÔME D'ETAT              53.0   
184  184  12145  1997-03-29    F  DIPLÔME D'ETAT              64.0   
185  185  12146  1995-09-29    F  DIPLÔME D'ETAT              58.0   
186  186  12148  1997-01-01    H  DIPLÔME D'ETAT              56.0   
187  187  12164  1992-10-29    H  DIPLÔME D'ETAT              55.0   
188  188  12174  1992-05-27    H  DIPLÔME D'ETAT              50.0   
189  189  12175 

##### Missing values checking 

In [52]:
DatasetSMF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 194 entries, 0 to 193
Data columns (total 26 columns):
ID                  194 non-null int64
ID                  194 non-null int64
BIRTHDAY            194 non-null object
NAME                194 non-null object
DIPLOMTYPE          194 non-null object
DIPLOMPERCENTAGE    194 non-null float64
DIPLOMSECTION       194 non-null object
DIPLOMOPTION        194 non-null object
SCHOOL              194 non-null object
SCHOOLPROVINCE      194 non-null object
SCHOOLSTATUS        194 non-null object
MENT1-2ndYear       194 non-null object
MENT1-1year         194 non-null object
OPT-first           194 non-null object
PROM-START          194 non-null object
PROM-END            194 non-null object
MENT2-2ndYear       194 non-null object
MENT2-1year         194 non-null object
PERC1-2ndYear       90 non-null float64
PERC1-1year         90 non-null float64
FAC-first           194 non-null object
PERC2-2ndYear       163 non-null float64
PERC2-1year    

with that command we discover that only Four features has missing values all of them will be filled by 0

In [63]:
DatasetSMF['PERC1-2ndYear']=DatasetSMF['PERC1-2ndYear'].fillna(0)
DatasetSMF['PERC2-1year']=DatasetSMF['PERC2-1year'].fillna(0)
DatasetSMF['PERC2-2ndYear']=DatasetSMF['PERC2-2ndYear'].fillna(0)
DatasetSMF['PERC1-1year']=DatasetSMF['PERC1-1year'].fillna(0)

In [65]:
DatasetSMF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 194 entries, 0 to 193
Data columns (total 26 columns):
ID                  194 non-null int64
ID                  194 non-null int64
BIRTHDAY            194 non-null object
NAME                194 non-null object
DIPLOMTYPE          194 non-null object
DIPLOMPERCENTAGE    194 non-null float64
DIPLOMSECTION       194 non-null object
DIPLOMOPTION        194 non-null object
SCHOOL              194 non-null object
SCHOOLPROVINCE      194 non-null object
SCHOOLSTATUS        194 non-null object
MENT1-2ndYear       194 non-null object
MENT1-1year         194 non-null object
OPT-first           194 non-null object
PROM-START          194 non-null object
PROM-END            194 non-null object
MENT2-2ndYear       194 non-null object
MENT2-1year         194 non-null object
PERC1-2ndYear       194 non-null float64
PERC1-1year         194 non-null float64
FAC-first           194 non-null object
PERC2-2ndYear       194 non-null float64
PERC2-1year  

now we can see that all missing values in our dataset are filled by 0

Alternatively, you can also choose to interpolate missing values: the interpolate() function will perform a linear interpolation at the missing data points to “guess” the value that is most likely to be filled in.
df.interpolate()